File: 2ndPeriod/aedsII/sort/QuickSort.cs
using System;

class MainClass {

    // Define o tamanho para o vetor
    const int TAM = 7;

    /*
    * Realiza a troca de posição de alguns elementos
    */
    public static void swap(int i, int j, int [] array) {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }

    /*
    * Método QuickSort de ordenação
    */
    private static void QuickSort(int esq, int dir, int [] array) {
        int i = esq, j = dir;
        int pivo = array[(dir+esq)/2];
        while (i <= j) {
            while (array[i] < pivo) i++;
            while (array[j] > pivo) j--;
            if (i <= j) {
                swap(i, j, array);
                i++;
                j--;
            }
        }
        if (esq < j)  QuickSort(esq, j, array);
        if (i < dir)  QuickSort(i, dir, array);
    }

    /*
     *
     */
    public static void Main (string[] args) {
        int[] vetor = new int[TAM]{20, 10, 30, 50, 45, 35, 5};

        
        for(int i=0; i<TAM; i++)
            Console.Write(vetor[i] + " ");
        Console.Write("\n\n");

        QuickSort(0, TAM -1, vetor);

        for(int i=0; i<TAM; i++)
            Console.Write(vetor[i] + " ");
        Console.Write("\n");
    }
}

File: 2ndPeriod/aedsII/sort/quicksort.c
#include <stdio.h>

#define TAM 7

void Swap(int i, int j, int *array)
{
    int tmp = array[i];
    array[i] = array[j];
    array[j] = tmp;
}

void QuickSort(int left, int right, int *array)
{
    int i = left, j = right, pivot = array[(right+left)/2];
    while(i <= j)
    {
        while(array[i] < pivot) i++; //Lower half
        while(array[j] > pivot) j--; //Higher half
        if(i <= j)
        {
            Swap(i, j, array);
            i++;
            j--;
        }
    }
    if(left < j) QuickSort(left, j, array); //Recursive for lower half 
    if(right > i) QuickSort(i, right, array); //Recursive for higher half
}

int main(void)
{
    int array[TAM] = {0, 1, 5, 3, 15, 16, 9, 10, 4, 3, 30, 5, 20, 48, 71, 82};
 
    for(int i=0; i<TAM; i++)
        printf("%d ",array[i]);
    printf("\n\n");

    QuickSort(0, TAM -1, array);

    for(int i=0; i<TAM; i++)
        printf("%d ",array[i]);
    printf("\n");
    
    return 0;
}